slang-antlr/src/test/java/com/sonarsource/slang/antlr/SLangConverterTest.java

```diff
@@ -99,7 +102,12 @@ public void additive_expression_with_multiple_operands() {
   public void function() {
     FunctionDeclarationTree function = parseFunction("private int foo(x1, x2) { x1 + x2 }");
     assertThat(function.name().name()).isEqualTo("foo");
-    // assertTree(function.returnType()).isIdentifier("boolean");
+    NativeKind expectedModifierKind = new SNativeKind(SLangParser.MethodModifierContext.class, "private");
+    assertThat(function.modifiers()).hasSize(1);
+    Tree modifierTree = function.modifiers().get(0);
+    assertTree(modifierTree).isInstanceOf(NativeTree.class);
+    assertThat(((NativeTree) modifierTree).nativeKind()).isEqualTo(expectedModifierKind);
```

Comment:
Why do we need that? I would prefer unit tests to show me why need different native kinds in some cases: in other words, its impact on syntactic equivalence. So far, I don't get why we need differentiatorObjs for slang.

Reply:
@pynicolas I agree. But `SyntacticEquivalence` is in `slang-checks` module. So if it's fine with you, I'll move it into api module to do this?

Reply:
Fine for me. Although I'm not sure it's easy...